### DISCUSSION TITLE
This is a fantastic library ... Thanks so much!

### DISCUSSION THREAD

**Discussion by kodell80305** [rzeldent/esp32-smartdisplay#141] [General]
It's a great example of how to do software supporting different hardware right - I've looked at dozen's of examples of LVGL libraries and projects people have implemented - this is the only one that keeps support for new boards simple.   I not only got my CYD (the  esp32-2432s028-v3) working, but also was able to create a new board type for the MKS DLC32 TS35-R display and use it.   I've been struggling with this display as there isn't much information about it.

**Comment by rzeldent** [comment]
Then star it!

If you have a working new board definition, I can add it to the repository, maybe easier for you also?!

**Comment by kodell80305** [comment]
Absolutely - I'll attach it.

I wasn't sure if this was in line with your project goals.

I would argue that the MKS DLC32 might be of interest to the community of
CYD users.   You can buy the board, the TS35-R display and 3 TMC2209
stepper drivers shipped on aliexpress for $35 US.   It has a nice 5.5 mm
barrel connector for power (cheap LED power bricks, also good for
automotive, RV, boat applications.   Also has I2C, a 12/24 volt PWM, a
bunch of 5 volt inputs & a  few outputs.  Everything on nice connectors &
good for solenoid/servo/stepper.    A little more expensive than a CYD ...
but still pretty reasonable.   I actually started with the CYD, but
realized that having the power/io made the project actually cheaper/more
robust.

There is a 2.4 inch version of this screen the TS24-4 that I suspect is
pretty similar - I haven't tried it as the price differential is so small -
it's only about $4 cheaper.

I started by taking the MKS DLC32 fork of grpl32 - trying to extract the
relevant code was a nightmare.   They've included many of their own header
files and their code into the LVGL and TFT-eSPI libraries (I can understand
if they need to extend some of the base functions, but at a minimum they
should have put some of this code into a new file, or used derived classes
or even ifdef's.

No guarantees on the board file, but it works for the demo & I've managed
to use the SquareLine software to replace the UI files with my own and make
everything work (touch screen, display).

Thanks!

Kevin M. O'Dell


On Sun, Mar 10, 2024 at 2:24 PM Rene ***@***.***> wrote:

> If you have a working new board definition, I can add it to the
> repository, maybe easier for you also?!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rzeldent/esp32-smartdisplay/discussions/141#discussioncomment-8737962>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BFP254QHVSDNFSX7DVDVZRDYXS6RPAVCNFSM6AAAAABEPF6MO2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DOMZXHE3DE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Kevin O'Dell
1717 View Point
Boulder, CO 80305

**Comment by kodell80305** [comment]
So the file I sent you only works with the DLC32 & TS35-R screen.

I'm wondering if including it is beyond the scope of your project which is
oriented toward the Sunton board & clones - you've already got a pretty
impressive list of boards just from that.

If you start including all the possible combinations of screens, boards &
that they may be hooked together in various ways it may not be scalable
(this is sort of what platformio/arduino do when they want you to specify a
board) - there are a huge number of variations & none of them seem to
apply.   I'm pretty new to the esp32/platformio/arduino world, so I'm not
sure what the solution is.   You almost want a web page that would generate
a board file & that you could tweak if needed.   What I've found
frustrating is that I've had to pick some more-or-less appropriate board
without knowing what it was really doing.

Anyway, feel free to do whatever you think is appropriate - I owe you
several hours of my life & possibly my sanity.

Kevin

On Sun, Mar 10, 2024 at 3:22 PM Kevin Odell ***@***.***> wrote:

> Absolutely - I'll attach it.
>
> I wasn't sure if this was in line with your project goals.
>
> I would argue that the MKS DLC32 might be of interest to the community of
> CYD users.   You can buy the board, the TS35-R display and 3 TMC2209
> stepper drivers shipped on aliexpress for $35 US.   It has a nice 5.5 mm
> barrel connector for power (cheap LED power bricks, also good for
> automotive, RV, boat applications.   Also has I2C, a 12/24 volt PWM, a
> bunch of 5 volt inputs & a  few outputs.  Everything on nice connectors &
> good for solenoid/servo/stepper.    A little more expensive than a CYD ...
> but still pretty reasonable.   I actually started with the CYD, but
> realized that having the power/io made the project actually cheaper/more
> robust.
>
> There is a 2.4 inch version of this screen the TS24-4 that I suspect is
> pretty similar - I haven't tried it as the price differential is so small -
> it's only about $4 cheaper.
>
> I started by taking the MKS DLC32 fork of grpl32 - trying to extract the
> relevant code was a nightmare.   They've included many of their own header
> files and their code into the LVGL and TFT-eSPI libraries (I can understand
> if they need to extend some of the base functions, but at a minimum they
> should have put some of this code into a new file, or used derived classes
> or even ifdef's.
>
> No guarantees on the board file, but it works for the demo & I've managed
> to use the SquareLine software to replace the UI files with my own and make
> everything work (touch screen, display).
>
> Thanks!
>
> Kevin M. O'Dell
>
>
> On Sun, Mar 10, 2024 at 2:24 PM Rene ***@***.***> wrote:
>
>> If you have a working new board definition, I can add it to the
>> repository, maybe easier for you also?!
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/rzeldent/esp32-smartdisplay/discussions/141#discussioncomment-8737962>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BFP254QHVSDNFSX7DVDVZRDYXS6RPAVCNFSM6AAAAABEPF6MO2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DOMZXHE3DE>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***
>> .com>
>>
>
>
> --
> Kevin O'Dell
> 1717 View Point
> Boulder, CO 80305
>


-- 
Kevin O'Dell
1717 View Point
Boulder, CO 80305

**Reply by rzeldent** [reply]
Hi Kevin,

Attachments are probably stripped. Think it's better to make a gist: https://gist.github.com/

**Comment by kodell80305** [comment]
Hopefully this is the right way to do it:

https://gist.github.com/kodell80305/40f6b7d8a5a735c82eb7dbbf18013e91.js
